### ISSUE TITLE
Error #15: Initializing libiomp5md.dll,

### ISSUE THREAD

**Issue by Bstrum36** [ultralytics/hub#1157]
### Search before asking

- [x] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

Seems to be crashing after inference before rendering

from ultralytics import YOLO
import cv2
import matplotlib.pyplot as plt
import sys
from inspect import currentframe

import FreeSimpleGUI as sg

import torch
import torchvision
print("Packages Installed")

def ln():
    cf = currentframe()
    return cf.f_back.f_lineno

if torch.cuda.is_available():
    print("PyTorch found CUDA.")
    print(f"Number of GPUs available: {torch.cuda.device_count()}")
    for i in range(torch.cuda.device_count()):
        print(f"GPU {i}: {torch.cuda.get_device_name(i)}")
else:
    print("PyTorch could not find CUDA.")
# Load the model (Ensure you're using the correct model path)
filename = sg.popup_get_file('Will not see this message',file_types=(('jpg','*.jpg'),), no_window=True)
print(ln(),filename)

model = YOLO("C:/Users/User/Project1/yolov12s.pt")
#results = model.predict("https://ultralytics.com/images/bus.jpg", conf=0.5)
results = model.predict(filename, conf=0.5)
# Visualize predictions
result_image = results[0].plot()
# Save using OpenCV
cv2.imwrite("output.jpg", result_image)
# Optionally: Show the image

plt.imshow(result_image)
plt.axis('off')
plt.show()
sys.exit()

C:\Users\User\.conda\envs\AI_Test\python.exe C:\Users\User\PycharmProjects\PythonProject3\Ultra_test_01.py 
Packages Installed
PyTorch found CUDA.
Number of GPUs available: 1
GPU 0: NVIDIA GeForce RTX 3080 Ti
26 C:/Users/User/Project1/soccer.jpg

image 1/1 C:\Users\User\Project1\soccer.jpg: 448x640 6 persons, 1 sports ball, 50.5ms
Speed: 1.8ms preprocess, 50.5ms inference, 49.7ms postprocess per image at shape (1, 3, 448, 640)
OMP: Error #15: Initializing libiomp5md.dll, but found libiomp5md.dll already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://www.intel.com/software/products/support/.
pybind11::handle::dec_ref() is being called while the GIL is either not held or invalid. Please see https://pybind11.readthedocs.io/en/stable/advanced/misc.html#common-sources-of-global-interpreter-lock-errors for debugging advice.
If you are convinced there is no bug in your code, you can #define PYBIND11_NO_ASSERT_GIL_HELD_INCREF_DECREF to disable this check. In that case you have to ensure this #define is consistently used for all translation units linked into a given pybind11 extension, otherwise there will be ODR violations. The failing pybind11::handle::dec_ref() call was triggered on a EnumType object.

Process finished with exit code -1073740791 (0xC0000409)

Please advise

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bstrum36, thank you for raising an issue about [Ultralytics HUB](https://hub.ultralytics.com) 🚀! Please visit our [HUB Docs](https://docs.ultralytics.com/hub) to learn more:
- [**Quickstart**](https://docs.ultralytics.com/hub/quickstart). Start training and deploying YOLO models with HUB in seconds.
- [**Datasets: Preparing and Uploading**](https://docs.ultralytics.com/hub/datasets). Learn how to prepare and upload your datasets to HUB in YOLO format.
- [**Projects: Creating and Managing**](https://docs.ultralytics.com/hub/projects). Group your models into projects for improved organization.
- [**Models: Training and Exporting**](https://docs.ultralytics.com/hub/models). Train YOLOv5, YOLOv8, YOLO11, and YOLO12 models on your custom datasets and export them to various formats for deployment.
- [**Integrations**](https://docs.ultralytics.com/hub/integrations). Explore different integration options for your trained models, such as TensorFlow, ONNX, OpenVINO, CoreML, and PaddlePaddle.
- [**Ultralytics HUB App**](https://docs.ultralytics.com/hub/app). Learn about the Ultralytics App for iOS and Android, which allows you to run models directly on your mobile device.
    * [**iOS**](https://docs.ultralytics.com/hub/app/ios). Learn about YOLO CoreML models accelerated on Apple's Neural Engine on iPhones and iPads.
    * [**Android**](https://docs.ultralytics.com/hub/app/android). Explore TFLite acceleration on mobile devices.
- [**Inference API**](https://docs.ultralytics.com/hub/inference-api). Understand how to use the Inference API for running your trained models in the cloud to generate predictions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/), including screenshots and detailed steps to reproduce the problem. This helps us investigate and address the issue efficiently.

If this is a ❓ Question, please include all relevant information, such as your dataset, model type, environment details, and anything else that could help us assist you better.

This is an automated response 🤖—an Ultralytics engineer will review your issue and follow up with personalized assistance soon. Thank you for your patience!

**Comment by pderrenger** [comment]
Hi @Bstrum36! 👋 

Thanks for the detailed error report. You're encountering two related issues that seem to stem from OpenMP library conflicts and threading problems. Let me help you troubleshoot this step by step.

## Quick Fixes to Try

### 1. OpenMP Library Conflict (Error #15)
The primary issue is multiple OpenMP runtime libraries being loaded. As the error suggests, you can resolve this by setting an environment variable:

**Windows (Command Prompt/PowerShell):**
```bash
set KMP_DUPLICATE_LIB_OK=TRUE
python your_script.py
```

**Or add this at the very beginning of your Python script:**
```python
import os
os.environ['KMP_DUPLICATE_LIB_OK'] = 'TRUE'
```

### 2. Model Recommendation
I notice you're using `yolov12s.pt`. For optimal stability and performance, I'd recommend switching to **YOLO11**, which is our latest stable and recommended model:

```python
model = YOLO("yolo11s.pt")  # Instead of yolov12s.pt
```

YOLO11 offers better stability and doesn't have the training instability issues that can affect YOLO12.

## Verification Steps

Since this appears to be a potential bug, please help us verify by:

1. **Updating to the latest versions:**
   ```bash
   pip install --upgrade ultralytics torch torchvision
   ```

2. **Checking your current package versions:**
   ```bash
   yolo checks
   ```
   or
   ```bash
   pip list
   ```

3. **Testing with a minimal example** using YOLO11:
   ```python
   from ultralytics import YOLO
   import os
   
   # Set environment variable to handle OpenMP conflict
   os.environ['KMP_DUPLICATE_LIB_OK'] = 'TRUE'
   
   # Use YOLO11 instead
   model = YOLO("yolo11s.pt")
   results = model.predict("path/to/your/image.jpg", conf=0.5)
   
   # Simple result handling
   for result in results:
       result.show()  # Display results
   ```

## Additional Resources

If the issue persists after trying these fixes, please check out our [troubleshooting guide for common YOLO issues](https://docs.ultralytics.com/guides/yolo-common-issues/), which covers installation errors and model training issues.

For creating a detailed bug report if needed, our [Minimum Reproducible Example guide](https://docs.ultralytics.com/help/minimum-reproducible-example/) can help you structure the information effectively.

Let me know if the OpenMP environment variable fix resolves your issue, or if you need help with the updated YOLO11 model! 🚀

**Comment by Bstrum36** [comment]
On another topic..
Can you tell me where I can download object365 models (.pt files)
 yolo11n\_object365.pt  is mentioned but no references to download sites.

Thanks

**Comment by Bstrum36** [comment]
Thanks Paula,
I will try all your suggestions.

Here is a yolo12x output video that I think will leave little doubt that
this is the real deal!  I'm getting ~14 -18 ms inference times running this
on my I9 with a 4030 TI PC.

https://drive.google.com/file/d/1It6flPehxn9RNMEFEVfJGRZ7T6vqMpOF/view?usp=sharing

Carbon Based Bill

**Comment by Bstrum36** [comment]
Hi Paula (again)

https://huggingface.co/NRtred/yolo11n_object365/blob/main/yolo11n_object365.pt

This works. I could not find any of the references you gave,

CB Bill
